### PR TITLE
Add Union-style indexing to `.points`, `.multipoints`, `.lines`, and `.polygons` `GeoSeries` accessors

### PR DESCRIPTION
**Pull request by thomcom**
## Description
This PR uses the `column._meta` object to slice the sub-geometry accessors named in the title so that `.x`, `.y`, and `.xy` uses reflect any previous slicing that has occurred to the `GeoSeries`.

Fixes https://github.com/rapidsai/cuspatial/issues/683

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cuspatial/cuspatial/core/geoseries.py [hunk 1]
```diff
@@ -118,28 +120,50 @@ def type(self):
...
+            return cudf.Series(result[::2])
 
         @property
         def y(self):
-            return cudf.Series(self._col.leaves().values[1::2])
+            types = self._meta.input_types
+            offsets = self._meta.union_offsets
+            indices = offsets[types == self._type.value]
+            result = self._col.take(indices._column).leaves().values
+            return cudf.Series(result[1::2])
```
Comment (isVoid) [inline python/cuspatial/cuspatial/core/geoseries.py:143]:
```suggestion
            return self.xy[1::2]
```

File: python/cuspatial/cuspatial/core/geoseries.py [hunk 2]
```diff
@@ -118,28 +120,50 @@ def type(self):
...
+            self._type = Feature_Enum.POINT
 
         @property
         def x(self):
-            return cudf.Series(self._col.leaves().values[0::2])
+            types = self._meta.input_types
+            offsets = self._meta.union_offsets
+            indices = offsets[types == self._type.value]
+            result = self._col.take(indices._column).leaves().values
+            return cudf.Series(result[::2])
```
Comment (isVoid) [inline python/cuspatial/cuspatial/core/geoseries.py:135]:
```suggestion
            return self.xy[::2]
```

**Review by isVoid — APPROVED** [review]
Nice.

**Comment by thomcom** [comment]
@gpucibot merge

**Pull request merged**